src/NLog/Fluent/LogBuilder.cs

```diff
@@ -302,21 +310,12 @@ public LogBuilder StackTrace(StackTrace stackTrace, int userStackFrame)
             if (callerLineNumber != 0)
                 Property("CallerLineNumber", callerLineNumber);
 
-            _logEvent.SetCallerInfo(null, callerMemberName, callerFilePath, callerLineNumber);
+            if (callerMemberName != null || callerFilePath != null || callerLineNumber != 0)
```

Comment:
if one of them isnot set, then we ignore them all?

it think `||` should be `&&`

Reply:
No it says only call `SetCallerInfo` if one of them is NOT null. Better handle NetFramework version without support for caller-member-attributes (NET3_5 +NET4_0)